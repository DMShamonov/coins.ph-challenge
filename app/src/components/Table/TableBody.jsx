// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableBodyPropsType } from './types.js.flow';

export default function TableBody({
  className,
  children,
}: TableBodyPropsType): React.Node {
  return (
    <div className={classnames('table__body', className)}>
      {children}
    </div>
  );
}
