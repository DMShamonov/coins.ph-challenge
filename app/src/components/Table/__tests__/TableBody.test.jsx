import React from 'react';
import { shallow } from 'enzyme';

import TableBody from '../TableBody';

describe('`TableBody` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<TableBody>__CONTENT__</TableBody>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<TableBody className="__CLASSNAME__">__CONTENT__</TableBody>)).toMatchSnapshot();
  });
});
