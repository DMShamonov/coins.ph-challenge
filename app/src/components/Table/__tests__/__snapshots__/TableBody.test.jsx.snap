// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`TableBody\` component should render correctly with \`className\` prop 1`] = `
<div
  className="table__body __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`TableBody\` component should render correctly with default props 1`] = `
<div
  className="table__body"
>
  __CONTENT__
</div>
`;
